### PR TITLE
DOC: Cast to compare git vs datalad for a common submodule workflow

### PR DESCRIPTION
**Pull request by mih**
It would be instrumental to get feedback on whether I am missing a more elegant approach for Git. Thx!

Recorded asciicast at https://raw.githubusercontent.com/datalad/talk-fosdem-2020/master/asciicast/datalad_convenience_vs_git.json

### CODE REVIEW

File: docs/casts/datalad_convenience_vs_git.sh [hunk 1]
```diff
@@ -0,0 +1,109 @@
+say "Fast forward through some prior work..."
+
+say "Let's assume a student developed an algorithm, using Git."
+run "mkdir code; git -C code init; touch code/work; git -C code add work ; git -C code commit -m 'MSc thesis by student A done'"
```
Comment (yarikoptic) [inline docs/casts/datalad_convenience_vs_git.sh]: To be realistic, drop use of -C

Comment (mih) [reply docs/casts/datalad_convenience_vs_git.sh]: That is only the setup. The key bits are the "PI" work.

File: docs/casts/datalad_convenience_vs_git.sh [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+run "datalad clone --dataset . ../analysis study"
+say "Requesting a particular file automatically obtains all needed subdatasets"
+run "datalad get study/code/work"
+
+say "Apply the code fix, and have it be detected"
+run "echo 'fix' >> study/code/work"
+run "datalad status"
+
+say "Unlike Git, DataLad makes nested datasets feel like a monorepo"
+run "datalad status --recursive --untracked all"
```
Comment (kyleam) [inline docs/casts/datalad_convenience_vs_git.sh]: `--untracked all` isn't relevant for this example and could dropped so there is less the viewer has to process.

File: docs/casts/datalad_convenience_vs_git.sh [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+run "git submodule update --init --recursive"
+
+say "Arrgh, there is a bug in the code, and the postdoc isn't here to work the fix in. But hey, Git is all distributed -- let's apply it right here, so we can move on with the science..."
+run "echo 'fix' >> study/code/work"
+
+say "Quickly commit the fix, so it can be pushed upstream later..."
+run "git add study/code/work"
+
+say "Erm... Needs to happen in the repository that actually contains the file...Need to look up the boundaries..."
+run "git submodule status --recursive"
```
Comment (kyleam) [inline docs/casts/datalad_convenience_vs_git.sh]: With  a mostly clean hierarchy, `git diff` also makes it fairly easy to spot the repo with the changed file:

```
$ git diff --submodule=diff
Submodule study contains modified content
Submodule code contains modified content
diff --git a/study/code/work b/study/code/work
index e69de29..a609090 100644
--- a/study/code/work
+++ b/study/code/work
@@ -0,0 +1 @@
+fix
```

File: docs/casts/datalad_convenience_vs_git.sh (hunk 1 above)
Comment (kyleam) [reply docs/casts/datalad_convenience_vs_git.sh]: [ a not relevant quip ]

> To be realistic,

It amused me to see this phrase near a snippet where the "algorithm" was an empty file.

File: docs/casts/datalad_convenience_vs_git.sh [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+say "Ok, let's do this..."
+run "cd study/code"
+run "git add work && git commit -m 'Fix'"
+
+say "Damn, the postdoc didn't specify a branch when adding the submodule. Need to fix, or it will be difficult to push."
```
Comment (kyleam) [inline docs/casts/datalad_convenience_vs_git.sh]: This is wrong, I think.  The branch recorded in .gitmodules doesn't change whether Git will get into a detached state.  It is used to identify the branch from the remote that should be used to update the submodule when `submodule update --remote [...]` is called.

File: docs/casts/datalad_convenience_vs_git.sh (hunk 3 above)
Comment (mih) [reply docs/casts/datalad_convenience_vs_git.sh]: Thanks! That is much more elegant.

File: docs/casts/datalad_convenience_vs_git.sh (hunk 4 above)
Comment (mih) [reply docs/casts/datalad_convenience_vs_git.sh]: ok, will reword.

File: docs/casts/datalad_convenience_vs_git.sh (hunk 2 above)
Comment (mih) [reply docs/casts/datalad_convenience_vs_git.sh]: Done

File: docs/casts/datalad_convenience_vs_git.sh (hunk 4 above)
Comment (mih) [reply docs/casts/datalad_convenience_vs_git.sh]: Done

File: docs/casts/datalad_convenience_vs_git.sh (hunk 3 above)
Comment (mih) [reply docs/casts/datalad_convenience_vs_git.sh]: Done

**Pull request merged**